simulation/src/simeeprom.cpp

```diff
@@ -22,15 +24,98 @@ void  EEPROMClass::write( int idx, uint8_t val )
 {
     printf("Writing to EEPROM address %u = %u\n", idx, val);
     someFakeEEPROM_memory[idx] = val;
+
+    if(autoUpdate && storage)
+    {
+        fseek(storage, idx, SEEK_SET);
+        fputc(val, storage);
+        fflush(storage);
+    }
```

Comment:
It might be better to do the update once every loop instead triggered by the simulator since there can be many writes to the EEPROM in a loop() call.

Reply:
Yes, you could certainly write the whole buffer to file after some period of inactivity, or once per loop() if it's flagged as dirty. But on the other hand a file write nowadays takes almost no time, and even if it did it would just mimic the actual EEPROM.write() call which takes a little bit of time.

Reply:
Also I added a fflush() in there just to have the "latest state" in case I crashed the whole thing without getting to the fclose()...

Reply:
I guess it depends on what the use-case is. The way you wrote it is probably best for the how you want to use it.